DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Utility.cs

```diff
@@ -0,0 +1,181 @@
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core
+{
+    /// <summary>
+    /// Utility functions are used to parse DICOM data including DA, DT, AS. The format for these VR is given in DICOM standard.
+    /// http://dicom.nema.org/medical/Dicom/2017e/output/chtml/part05/sect_6.2.html
+    /// </summary>
+    public static class Utility
+    {
+        public static readonly Dictionary<string, AgeType> AgeTypeMapping = new Dictionary<string, AgeType>
+            {
+                { "Y", AgeType.Year },
+                { "M", AgeType.Month },
+                { "W", AgeType.Week },
+                { "D", AgeType.Day },
+            };
+
+        public static DateTimeOffset[] ParseDicomDate(DicomDate item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return item.Get<string[]>().Select(ParseDicomDate).ToArray();
+        }
+
+        public static DateTimeOffset ParseDicomDate(string date)
+        {
+            EnsureArg.IsNotNull(date, nameof(date));
+
+            try
+            {
+                return DateTimeOffset.ParseExact(date, "yyyyMMdd", CultureInfo.InvariantCulture);
+            }
+            catch
+            {
+                throw new DicomDataException("Invalid date value. The valid format is YYYYMMDD.");
+            }
+        }
+
+        public static DateTimeObject[] ParseDicomDateTime(DicomDateTime item)
+        {
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            return item.Get<string[]>().Select(ParseDicomDateTime).ToArray();
+        }
+
+        public static DateTimeObject ParseDicomDateTime(string dateTime)
+        {
+            EnsureArg.IsNotNull(dateTime, nameof(dateTime));
+
+            Regex dateTimeRegex = new Regex(@"^((?<year>\d{4})(?<month>\d{2})(?<day>\d{2})(?<hour>\d{2})(?<minute>\d{2})(?<second>\d{2})(\.(?<millisecond>\d{1,6}))?(?<timeZone>(?<sign>-|\+)(?<timeZoneHour>\d{2})(?<timeZoneMinute>\d{2}))?)(\s*)");
+            var matches = dateTimeRegex.Matches(dateTime);
+            if (matches.Count != 1)
+            {
+                throw new DicomDataException("Invalid date time value. The valid format is YYYYMMDDHHMMSS.FFFFFF&ZZXX.");
+            }
+
+            var groups = matches[0].Groups;
+
+            int year = groups["year"].Success ? int.Parse(groups["year"].Value) : 0;
+            int month = groups["month"].Success ? int.Parse(groups["month"].Value) : 0;
+            int day = groups["day"].Success ? int.Parse(groups["day"].Value) : 0;
+            int hour = groups["hour"].Success ? int.Parse(groups["hour"].Value) : 0;
+            int minute = groups["minute"].Success ? int.Parse(groups["minute"].Value) : 0;
+            int second = groups["second"].Success ? int.Parse(groups["second"].Value) : 0;
+            int millisecond = groups["millisecond"].Success ? int.Parse(groups["millisecond"].Value) : 0;
+            millisecond = millisecond > 999 ? int.Parse(millisecond.ToString().Substring(0, 3)) : millisecond;
```

Comment:
>millisecond [](http://example.com/codeflow?start=98&length=11)

/ 1000

Reply:
Remember to modify the group name in regular expression. It should be microsecond as well.
Parse the first 3 digits directly should be OK.

Reply:
Thanks, fixed it by /1000